SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosOfferScheduler.java

```diff
@@ -118,33 +129,44 @@ public SingularityMesosOfferScheduler(MesosConfiguration mesosConfiguration, Cus
 
     while (!pendingTaskIdToTaskRequest.isEmpty() && addedTaskInLastLoop && canScheduleAdditionalTasks(taskCredits)) {
       addedTaskInLastLoop = false;
-      Collections.shuffle(offerHolders);
 
-      for (SingularityOfferHolder offerHolder : offerHolders) {
-        if (configuration.getMaxTasksPerOffer() > 0 && offerHolder.getAcceptedTasks().size() >= configuration.getMaxTasksPerOffer()) {
-          LOG.trace("Offer {} is full ({}) - skipping", offerHolder.getOffer(), offerHolder.getAcceptedTasks().size());
-          continue;
+      for (SingularityTaskRequestHolder taskRequestHolder : pendingTaskIdToTaskRequest.values()) {
+
+        Map<SingularityOfferHolder, Double> scorePerOffer = new HashMap<>();
+
+        for (SingularityOfferHolder offerHolder : offerHolders) {
+
+          if (configuration.getMaxTasksPerOffer() > 0 && offerHolder.getAcceptedTasks().size() >= configuration.getMaxTasksPerOffer()) {
+            LOG.trace("Offer {} is full ({}) - skipping", offerHolder.getOffer(), offerHolder.getAcceptedTasks().size());
+            continue;
+          }
+
+          double score = score(offerHolder, stateCache, tasksPerOfferPerRequest, taskRequestHolder, getUsagesPerRequestTypePerSlave());
+          if (score > 0) {
+            // todo: can short circuit here if score is high enough
+            scorePerOffer.put(offerHolder, score);
```

Comment:
Thought we might want to have a value that's definitely good enough to just accept instead of continue evaluating